cmsis/README.md

```diff
@@ -0,0 +1,46 @@
+
+# CMSIS Version 5
+
+This directory contains a ported version of the upstream repository for [CMSIS_5](http://arm-software.github.io/CMSIS_5/General/html/index.html) as well as Mbed specific RTX configuration and RTOS boot source files.
+
+The upstream source code is imported using the Python script located at [`tools/importer/importer.py`](../tools/importer/importer.py) when passed the [`tools/importer/cmsis_importer.json`](../tools/importer/cmsis_importer.json) configuration file.
+For more information on the importer script please read [`tools/importer/README.md`](../tools/importer/README.md).
+
+The directory is organised as follows:
+
+```
++--cmsis
+|   +-- README.md                           # The present mardown file
+|   +-- TARGET_CORTEX_A/
+|       +-- TOOLCHAIN_IAR/
+|           +-- cmain.S                     # Custom IAR toolchain non-RTOS boot sequence for Mbed
+|   +-- device/
+|       +-- rtos/
+|           +-- mbed_lib.json               # Mbed library configuration file
+|           +-- include/
+|               +-- mbed_boot.h             # Declares the functions that run before main()
+|               +-- mbed_cmsis_conf.h       # Process stack configuration
+|               +-- mbed_rtx_conf.h         # Changes to RTX configuration
+|               +-- mbed_rtx_storage.h      # Declares the primitives storage types for RTX
+|           +-- source/
+|               +-- mbed_boot.c             # Implements some of the functions that run before main()
+|               +-- mbed_rtos_rtx.c         # Implements some of the functions that run before main()
+|               +-- mbed_rtx_handlers.c     # Implements RTX handlers
+|               +-- mbed_rtx_idle.cpp       # Implements RTX idle loop
+|           +-- RTE/
+|               +-- include/
+|                   +-- RTE_Components.h    # Run-Time-Environment Component Configuration File
```

Comment:
Does RTE_Components.h belong under `rtos`, it configures cmsis as well. Considering `include` we follow, should top cmsis component have include path that would actually have this component header that applies to all components within cmsis?

Reply:
The `RTE` [has been moved](https://github.com/ARMmbed/mbed-os/compare/5c6eb9768a10e18a26c8da114f8b21744c113f68..d7856c4c56711354da2ae038950b4f1fc97f2776).